src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -120,7 +127,13 @@ public FSDataOutputStream append(Path f, int bufferSize, Progressable progress)
      */
     @Override
     public boolean rename(Path src, Path dst) throws IOException {
-        return false;
+        FileSystemPathTuple srcTuple = generateFSPathTuple(src);
+        FileSystemPathTuple dstTuple = generateFSPathTuple(dst);
+        if(!srcTuple.getFileSystem().equals(dstTuple.getFileSystem())) {
+            LOG.warn("Cannot rename paths from different FileSystems");
```

Comment:
```suggestion
             LOG.warn("Cannot rename between different underlying FileSystems");
```
and ideally include the names of the filesystems.